nav2_recoveries/src/spin.cpp

```diff
@@ -87,13 +88,10 @@ Status Spin::onCycleUpdate()
   geometry_msgs::msg::Pose2D pose2d;
   pose2d.x = current_pose.position.x;
   pose2d.y = current_pose.position.y;
-  pose2d.theta = tf2::getYaw(current_pose.orientation) +
-    cmd_vel.angular.z * (1 / cycle_frequency_);
+  pose2d.theta = current_yaw;
 
-  if (!collision_checker_->isCollisionFree(pose2d)) {
-    stopRobot();
-    RCLCPP_WARN(node_->get_logger(), "Collision Ahead - Exiting Spin ");
-    return Status::SUCCEEDED;
+  if (!isCollisionFree(yaw_diff, cmd_vel, pose2d)) {
+    return Status::FAILED;
```

Comment:
Here it also applies the comments from `BackUp`, return `SUCCEEDED` and stop the robot here rather than inside `isCollisionFree`.

Reply:
Commit to follow